packages/context/src/inject.ts

```diff
@@ -192,21 +200,7 @@ export function describeInjectedProperties(
   // tslint:disable-next-line:no-any
   target: any,
 ): {[p: string]: Injection} {
-  const metadata: {[name: string]: Injection} = {};
-  let obj = target;
-  while (true) {
-    const m = Reflector.getOwnMetadata(PROPERTIES_KEY, obj);
-    if (m) {
-      // Adding non-existent properties
-      for (const p in m) {
-        if (!(p in metadata)) {
-          metadata[p] = m[p];
-        }
-      }
-    }
-    // Recurse into the prototype chain
-    obj = Object.getPrototypeOf(obj);
-    if (!obj) break;
-  }
+  const metadata: {[name: string]: Injection} =
+    Reflector.getMetadata(PROPERTIES_KEY, target) || {};
```

Comment:
Similarly here - can we use a (possibly new) helper from `@loopback/metadata` to access property-level information?

Also the old code was recursing through the prototype chain while this new code is not, does it mean that `Reflector.getMetadata` is taking care of that already? I would expect a flag to control whether prototypes are consulted for metadata. If there is one (and is set to true by default), then please be more explicit here and set the flag to that default value, to make the intent more obvious to people reading this code for the first time.